serde/src/de/mod.rs

```diff
@@ -1732,6 +1732,25 @@ pub trait SeqAccess<'de> {
         self.next_element_seed(PhantomData)
     }
 
+    /// Equivalent to calling `next_element`, and then if there are no more remaining items,
+    /// calling `Error::invalid_length`.
+    ///
+    /// This method exists purely to reduce the size of the generated code,
+    /// reducing compile times. `SeqAccess` implementations should not override
```

Comment:
Is there any reason to implement is specifically as a method rather than a free function? Using a free function gives the desired no-overriding guarantee, and method call syntax is not used in derived code anyway.

Reply:
A method was easier, being more similar to the original function. But it's not fundamental.

Reply:
I agree with @AnthonyMikh here, that this would be better as free functions in `serde::__private::de` module